### PR TITLE
[CPU] Execute shape_of path in FP32 for BF16 case

### PR DESCRIPTION
**Pull request by mandrono**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *78144*

### CODE REVIEW

**Comment by mandrono** [comment]
need tests

File: src/plugins/intel_cpu/src/mkldnn_graph.cpp [hunk 1]
```diff
@@ -1279,6 +1279,80 @@ void MKLDNNGraph::EnforceBF16() {
             }
         }
     }
+
+    // in case of dynamism we need execute all shape_of path in FP32 to avoid possible accuracy loss
```
Comment (mandrono) [inline src/plugins/intel_cpu/src/mkldnn_graph.cpp:1283]: 78144

**Comment by dmitry-gorokhov** [comment]
@alexey-varyzgin could you please review the PR?

File: src/plugins/intel_cpu/src/mkldnn_graph.cpp [hunk 2]
```diff
@@ -1279,6 +1279,80 @@ void MKLDNNGraph::EnforceBF16() {
...
+        for (size_t i = 0; i < node->getParentEdges().size(); i++) {
+            for (size_t i = 0; i < node->getOriginalInputsNumber(); i++) {
+                if (node->getOriginalInputPrecisionAtPort(i) == Precision::BF16)
+                    node->setOriginalInputPrecisionAtPort(i, Precision::FP32);
+            }
+
+            for (size_t i = 0; i < node->getOriginalOutputsNumber(); i++) {
+                if (node->getOriginalOutputPrecisionAtPort(i) == Precision::BF16)
+                    node->setOriginalOutputPrecisionAtPort(i, Precision::FP32);
+            }
```
Comment (mandrono) [inline src/plugins/intel_cpu/src/mkldnn_graph.cpp:1310]: extract from cycle

**Comment by mandrono** [comment]
@alexey-varyzgin could you take a look please?

**Review by alexey-varyzgin — APPROVED** [review]

**Comment by mandrono** [comment]
@dmitry-gorokhov could you take a look please?

**Comment by maxnick** [comment]
Changes are cherry picked to PR https://github.com/openvinotoolkit/openvino/pull/14911

**Pull request closed without merging**